AIFG-3D-Connect-4/SfmlInput.cpp

```diff
@@ -0,0 +1,59 @@
+#include "SfmlInput.h"
+#include <iostream>
+///////////////////////////////////////////////////////////////////////////////
+SfmlInput::SfmlInput(GameBoard& t_gameBoard, std::vector<Grid>& t_grids) : m_gameBoard(t_gameBoard), m_grids(t_grids)
+{
+}
+
+///////////////////////////////////////////////////////////////////////////////
+Coordinate SfmlInput::getCoordinate()
+{
+	sf::Event newEvent;
+	Coordinate pos = Coordinate(-1, -1, -1 ); // Default value if no key/mouse is pressed
+	while (Window::getWindow().pollEvent(newEvent))
+	{
+		if (sf::Event::Closed == newEvent.type) // window message
+		{
+			Window().getWindow().close();
+		}
+		else if (sf::Event::KeyPressed == newEvent.type)
+		{
+			processKeys(newEvent);
+		}
+		else if (sf::Event::MouseButtonPressed == newEvent.type)
+		{
+			sf::Vector2f mousePos = Window::getWindow().mapPixelToCoords(sf::Mouse::getPosition(Window::getWindow()));
+			for (int i = 0; i < m_grids.size(); i++)
+			{
+				// Gets the cells within a grid.
+				std::vector<sf::RectangleShape> rects = m_grids.at(i).getRects();
```

Comment:
Along with the return value of the function, this should be stored in a const reference